File: TheaterSuggestions/CSharp/SeatsSuggestions/SeatingOptionSuggested.cs
using System.Collections.Generic;

namespace SeatsSuggestions;

public class SeatingOptionSuggested
{
    public SeatingOptionSuggested(SuggestionRequest suggestionRequest)
    {
        PartyRequested = suggestionRequest.PartyRequested;
        PricingCategory = suggestionRequest.PricingCategory;
    }

    public PricingCategory PricingCategory { get; }
    public List<SeatingPlace> Seats { get; } = new();
    public int PartyRequested { get; }

    public void AddSeatingPlace(SeatingPlace seatingPlace)
    {
        Seats.Add(seatingPlace);
    }

    public bool MatchExpectation()
    {
        return Seats.Count == PartyRequested;
    }
}

File: TheaterSuggestions/Java/SeatsSuggestions/src/main/java/com/baasie/SeatsSuggestions/SeatingOptionSuggested.java
package com.baasie.SeatsSuggestions;

import java.util.ArrayList;
import java.util.List;

public class SeatingOptionSuggested {

    private PricingCategory pricingCategory;
    private List<SeatingPlaces> seats = new ArrayList<>();
    private int partyRequested;

    public SeatingOptionSuggested(int partyRequested, PricingCategory pricingCategory) {
        this.pricingCategory = pricingCategory;
        this.partyRequested = partyRequested;
    }

    public void addSeat(SeatingPlaces seat) {
        seats.add(seat);
    }

    public boolean matchExpectation() {
        return seats.size() == partyRequested;
    }

    public List<SeatingPlaces> seats() {
        return seats;
    }
}
